src/Auth/EnvMomentoTokenProvider.php

```diff
@@ -13,17 +12,41 @@ class EnvMomentoTokenProvider implements ICredentialProvider
     private string $authToken;
     private string $controlEndpoint;
     private string $cacheEndpoint;
+    private ?string $trustedControlEndpointCertificateName = null;
+    private ?string $trustedCacheEndpointCertificateName = null;
 
-    public function __construct(string $envVariableName)
+    public function __construct(
+        string  $envVariableName,
+        ?string $controlEndpoint = null,
+        ?string $cacheEndpoint = null,
+        ?string $trustedControlEndpointCertificateName = null,
+        ?string $trustedCacheEndpointCertificateName = null
+    )
     {
         $authToken = getenv($envVariableName);
         if ($authToken === false || isNullOrEmpty($authToken)) {
             throw new InvalidArgumentError("Environment variable $envVariableName is empty or null.");
         }
-        $payload = AuthUtils::parseAuthToken($authToken);
         $this->authToken = $authToken;
-        $this->controlEndpoint = $payload->cp;
-        $this->cacheEndpoint = $payload->c;
+
+        $endpointArgs = [
+            $controlEndpoint, $cacheEndpoint, $trustedCacheEndpointCertificateName, $trustedControlEndpointCertificateName
+        ];
+        if ($this->anyAreDefined($endpointArgs)) {
+            if (!$this->allAreDefined($endpointArgs)) {
+                throw new InvalidArgumentError(
+                    "If any of controlEndpoint, cacheEndpoint, trustedCacheEndpointCertificateName, or " .
+                    "trustedControlEndpointCertificateName are provided, they must all be.");
```

Comment:
you can address this in a follow-up pr but why would we disallow the first two being passed by themselves?

Reply:
Good point.